src/gafaelfawr/database.py

```diff
@@ -29,10 +32,26 @@ def initialize_database(config: Config) -> None:
     config : `gafaelfawr.config.Config`
         The Gafaelfawr configuration.
     """
-    engine = create_engine(config.database_url)
-    initialize_schema(engine)
+    logger = structlog.get_logger(config.safir.logger_name)
+
+    # Check connectivity to the database and retry if needed.  This uses a
+    # pre-ping to ensure the database is available and attempts to connect
+    # five times with a two second delay between each attempt.
+    for _ in range(5):
+        try:
+            engine = create_engine(config.database_url, pool_pre_ping=True)
+            initialize_schema(engine)
+        except OperationalError:
+            logger.info("database not ready, waiting two seconds")
+            time.sleep(2)
+            continue
+        logger.info("initialized database schema")
+        break
+
```

Comment:
Is it worth logging an explicit error if the database connection never worked?

Reply:
Oh, good idea, will add.